web.rb

```diff
@@ -147,6 +147,30 @@ def validateApiKey
   return {:intent => payment_intent.id, :secret => payment_intent.client_secret}.to_json
 end
 
+# This endpoint creates a SetupIntent.
+# https://stripe.com/docs/api/setup_intents/create
+post '/create_setup_intent' do
+  validationError = validateApiKey
+  if !validationError.nil?
+    status 400
+    return log_info(validationError)
+  end
+
+  begin
+    setup_intent = Stripe::SetupIntent.create({
+      :payment_method_types => params[:payment_method_types] || ['card_present'],
```

Comment:
It might be a good idea to also support a few more commonly used parameters here:
* `customer`
* `description`
* `payment_method`

Reply:
+1 to what bibek suggested, I think `customer` param is very useful as we will attach the generated `card` PM on `customer` automatically on the backend.

Reply:
Added the new params!